### PR TITLE
add plugin transformer to AMP.extension call for singlepass

### PR DESCRIPTION
**Pull request by erwinmombay**
replace the `inlineAmpExtensionCall` found in https://github.com/ampproject/amphtml/blob/master/build-system/runner/src/org/ampproject/AmpPass.java#L155

### CODE REVIEW

**Comment by rsimha** [comment]
Thanks for sending this out. It will (at least partially) unblock #22056.

Once this is in, I'm not sure if the lack of these other passes will affect single-pass. Will re-test after this is merged.

https://github.com/ampproject/amphtml/blob/9d6477b4b638f96c52453d7c19492b693e561bdb/build-system/runner/src/org/ampproject/AmpPass.java#L75-L95

File: build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          const func = node.arguments[node.arguments.length - 1];
+          if (!(t.isFunctionExpression(func) ||
+              t.isArrowFunctionExpression(func))) {
+            return;
+          }
+
+          const newParams = [t.identifier('AMP')];
+          const newFunc = t.isFunctionExpression(func) ?
+              t.functionExpression(null, newParams, func.body) :
+              t.arrowFunctionExpression(newParams, func.body);
```
Comment (erwinmombay) [inline build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: @jridgewell i forgot to handle arrow expressions. added here and tests

File: build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  return {
+    name: 'transform-amp-extension-call',
+    visitor: {
+      CallExpression(path) {
+        const {node} = path;
+        const {callee} = node;
+        if (t.isIdentifier(callee.object, {name: 'AMP'}) &&
+            t.isIdentifier(callee.property, {name: 'extension'})) {
+          const func = node.arguments[node.arguments.length - 1];
+          if (!(t.isFunctionExpression(func) ||
```
Comment (jridgewell) [inline build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: I don't think this check is necessary?

File: build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js (hunk 1 above)
Comment (jridgewell) [reply build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: Why not just take invoke the original function? There's no need to create a new function node.

Comment (jridgewell) [reply build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: See https://astexplorer.net/#/gist/ed168eb5b5098c06a751208884ebff46/7a5af9903baa4817f26d079bf720ae82ad3a2899

File: build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js (hunk 2 above)
Comment (erwinmombay) [reply build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: removed

File: build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js (hunk 1 above)
Comment (erwinmombay) [reply build-system/babel-plugins/babel-plugin-transform-amp-extension-call/index.js]: thanks, done.  I wasn't sure if it was safe to reparent nodes like that.

File: build-system/runner/src/org/ampproject/AmpCommandLineRunner.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ protected AmpCommandLineRunner(String[] args) {
     CompilerOptions options = super.createOptions();
     options.setCollapsePropertiesLevel(CompilerOptions.PropertyCollapseLevel.ALL);
     AmpPass ampPass = new AmpPass(getCompiler(), is_production_env, suffixTypes,
-        assignmentReplacements, prodAssignmentReplacements, amp_version);
+        assignmentReplacements, prodAssignmentReplacements, amp_version, single_file_compilation);
```
Comment (rsimha) [inline build-system/runner/src/org/ampproject/AmpCommandLineRunner.java]: Note to self: remove this in #22056

**Comment by rsimha** [comment]
Looks like `runner.jar` needs to be rebuilt with these changes.

**Pull request merged**